### PR TITLE
fix(date-time-picker): support i18n features

### LINKED ISSUE #2656: [DateTimePicker] support i18n features
### What package is this for?

- [ ] React
- [x] Angular

### Summary

DateTimePicker supports any i18n features to allow different languages.
When the user changes the language, it'll translate it accordingly. 

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an
upcoming release?

_NB: The core contributors will try to work with your timeline, but it's not
guaranteed. The earlier you make a request in advance of a desired delivery
date, the better!_

### Available extra resources

What resources do you have to assist this effort?

_Carbon add-ons for Watson IoT is a collaborative system. We encourage teams to
build components and submit them for integration._

### PR DESCRIPTION
**Pull request by sophiiae**
Closes #2656 

**Summary**

- summary_here

**Change List (commits, features, bugs, etc)**

- items_here

**Acceptance Test (how to verify the PR)**

- tests here

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [x] PR should link and close out an existing issue

### CODE REVIEW

File: packages/angular/src/date-time-picker/date-time-picker.component.ts [hunk 1]
```diff
@@ -254,23 +316,29 @@ export class DateTimePickerComponent implements OnChanges, OnInit {
     const range = this.dateTimeRanges.find((range) => range.key === rangeOrType);
     const [start, end] = range.getRange();
     // TODO: provide a way to customize this for g11n
-    const formatString = 'yyyy-M-d HH:mm';
+    const formatString = this.dateFormat + ' ' + this.timeFormat;
```
Comment (cal-smith) [inline packages/angular/src/date-time-picker/date-time-picker.component.ts:319]: Lets use a template string for this:

```suggestion
    const formatString = `${this.dateFormat} ${this.timeFormat}`;
```

File: packages/angular/src/date-time-picker/date-time-picker.component.ts [hunk 2]
```diff
@@ -254,23 +316,29 @@ export class DateTimePickerComponent implements OnChanges, OnInit {
...
+      endFormatted = this.batchText.NOW;
     }
-    return `${format(start, formatString)} to ${endFormatted}`;
+    return `${format(start, formatString)}${this.i18n.get().rangeSeparator}${endFormatted}`;
   }
 
   formatCustomRange() {
     // TODO: provide a way to customize this for g11n
-    const formatString = 'yyyy-M-d HH:mm';
+    const formatString = this.dateFormat + ' ' + this.timeFormat;
```
Comment (cal-smith) [inline packages/angular/src/date-time-picker/date-time-picker.component.ts:329]: Same template string comment here:

```suggestion
    const formatString = `${this.dateFormat} ${this.timeFormat}`;
```

**Review by cal-smith — COMMENTED** [review]
A couple suggestions, but overall looks good! Thanks for fixing this!

**Review by cal-smith — APPROVED** [review]
Awesome! LGTM!

**Review by Donisius — APPROVED** [review]
Looks good!

**Pull request merged**